app/modules/key_management/secret_manager.py

```
@@ -189,3 +204,68 @@ def delete_secret(
+            api_key = await APIKeyService.get_api_key(user["user_id"], db)
+            if api_key is None:
+                raise HTTPException(
+                    status_code=404,
+                    detail="No API key found for this user"
+                )
+            return {"api_key": api_key}
+        except Exception as e:
+            raise HTTPException(
+                status_code=500,
+                detail=f"Failed to retrieve API key: {str(e)}"
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)